File: leetcode/0102. Binary Tree Level Order Traversal/Solution.swift
// https://leetcode.com/problems/binary-tree-level-order-traversal/

public class TreeNode {
    public var val: Int
    public var left: TreeNode?
    public var right: TreeNode?
    public init(_ val: Int) {
        self.val = val
        self.left = nil
        self.right = nil
    }
}

class Solution {
    func levelOrder(_ root: TreeNode?) -> [[Int]] {
        var levels = [[Int]]()
        guard let root = root else {
            return levels
        }

        var q = [TreeNode]()
        q.append(root)

        var level = [Int]()
        while !q.isEmpty {
            for _ in 0..<q.count {
                let n = q.removeFirst()
                level.append(n.val)

                if let left = n.left {
                    q.append(left)
                }
                if let right = n.right {
                    q.append(right)
                }
            }

            levels.append(level)
            level = []
        }

        return levels
    }
}

File: leetcode/0102. Binary Tree Level Order Traversal/Solution.py
# https://leetcode.com/problems/binary-tree-level-order-traversal/

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        levels = []
        if not root:
            return levels
        
        q = collections.deque([root])
        
        level = []
        while q:
            c = len(q)
            for _ in range(c):
                n = q.popleft()
                level.append(n.val)

                if n.left:
                    q.append(n.left)
                if n.right:
                    q.append(n.right)

            levels.append(level)
            level = []
        
        return levels